Add LibCURL to TizenRT Platform

bhargava-cs opened the pull request:
- Support HTTP1.1 insecure methods

bhargava-cs commented:
@juitem, @sunghan-chang, @jeongarmy
Addressed all CURL review comments related to previous PR.

bhargava-cs commented:
@jeongarmy,
artik053/nettest configuration is used to enable curl

sunghan-chang commented:
@bhargava-cs I am sorry but previous PR is better on commit side.
You must not include serveral changes on same commit.
Currently yours has adding example, adding libcurl core and etc.
They must be in different commit like your previous PR.
```
git reset --soft HEAD^
make them as unstage status
git add <core only>
git commit
git add <example>
git commit
....
git push -f
```

bhargava-cs commented:
@sunghan-chang 
Thanks for the steps. Split the patch into core and application.

sunghan-chang commented on external/curl/Make.defs:
```diff
@@ -0,0 +1,22 @@
+###########################################################################
+#
+# Copyright 2016 Samsung Electronics All Rights Reserved.
```
2018

bhargava-cs replied:
@sunghan-chang,
Will do the changes on monday.

jeongarmy commented:
Please fix assert when processing as I said.
I tested it with artik053/nettest on artik053 as below.
1. connect wifi
2, get ip address
3. execute curl "curl http get google.com"

bhargava-cs commented:
@sunghan-chang 
Addressed review comments and pushed the changes.
Would you please guide how to fix this merge conflict.

source "$EXTERNALDIR/codecs/Kconfig"
<<<<<<< master
source "$EXTERNALDIR/curl/Kconfig"
=======
source "$EXTERNALDIR/gmock/Kconfig"
>>>>>>> master
source "$EXTERNALDIR/iotivity/Kconfig"

pradeepns85 commented on apps/examples/curl_test/Kconfig:
```diff
@@ -0,0 +1,15 @@
+#
+
+config EXAMPLES_CURLTEST
+        bool "curl test example"
```
Kindly Use Tab

bhargava-cs replied:
OK

pradeepns85 commented on apps/examples/curl_test/Kconfig:
```diff
@@ -0,0 +1,15 @@
+
+config USER_ENTRYPOINT
+	string
+        default "curl_main" if ENTRY_CURLTEST
```
same as above

bhargava-cs replied:
Ok

pradeepns85 commented on external/curl/Makefile:
```diff
@@ -0,0 +1,85 @@
+
+ASRCS =
+
+# CURL common & feature specific source files
```
If possible, You may have to Segregate these huge set of files based on common and feature specific.
It would have helped to understand the dependencies of files per feature.

bhargava-cs replied:
We will optimize curl in consecutive patches. In first release we are targeting basic functionality.

bhargava-cs commented:
@pradeepns85 
Could you help me with "external/Kconfig" conflict resolution.

bhargava-cs commented:
@sunghan-chang , @juitem , @jeongarmy 
Addressed all review comments till date. We are going to push nghttp2 patch in this week.
Please do the needful to merge this PR.
Would it be possible to resolve "external/Kconfig" conflict while merging ?

lokeshbv commented:
Hi @bhargava-cs ,
Follow the below steps to resolve conflicts.
```
git fetch <upstaream_name>
git rebase --onto <upstaream_name>/master HEAD~2
```

lokeshbv commented:
Hi @bhargava-cs ,
 Follow the below steps to resolve conflicts.
```
git fetch <upstaream_name>
git rebase --onto <upstaream_name>/master HEAD~2
```

pradeepns85 commented:
@bhargava-cs . Kindly re-base this file to latest code and resolve the merge conflict and then push changes.

bhargava-cs commented:
@sunghan-chang , @juitem , @jeongarmy,
Addressed the review comments. Would you please do the needful.

sunghan-chang commented on apps/examples/curl_test/curl_main.c:
```diff
@@ -0,0 +1,242 @@
+	argc = input->argc;
+	argv = input->argv;
+
+	if (strcmp(argv[1],"http") == 0 || strcmp(argv[1],"https") == 0) {
```
Please add a space after ','.
```
[1], "http") .... [1], "httpes") ...
```

sunghan-chang commented on apps/examples/curl_test/curl_main.c:
```diff
@@ -0,0 +1,242 @@
+	argv = input->argv;
+
+	if (strcmp(argv[1],"http") == 0 || strcmp(argv[1],"https") == 0) {
+		if (strcmp(argv[2],"get") == 0) {
```
same as above, please check all of this kind of code. There are many.

sunghan-chang commented:
@bhargava-cs I am sorry, but @jeongarmy is not today. Tomorrow she will check and get back to you with test result. Thank you.
